Earthfile

```diff
@@ -142,6 +145,36 @@ docker:
 		SAVE IMAGE --push "${i}"
 	END
 
+flatpak:
+  RUN apk add flatpak flatpak-builder
+  RUN flatpak remote-add --user --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo
+  RUN flatpak install flathub org.freedesktop.Platform//23.08 org.freedesktop.Sdk//23.08 -y
+
+  ARG TARGETPLATFORM   # provided by Earthly
+  IF [ "$TARGETPLATFORM" = "linux/amd64" ]
+    ARG NAME=linux-64bit
+  ELSE IF [ "$TARGETPLATFORM" = "linux/386" ]
+    ARG NAME=linux-32bit
+  ELSE IF [ "$TARGETPLATFORM" = "linux/arm64" ]
+    ARG NAME=linux-arm64
+  ELSE IF [ "$TARGETPLATFORM" = "linux/arm/v7" ]
+    ARG NAME=linux-arm7
+  ELSE IF [ "$TARGETPLATFORM" = "darwin/amd64" ]
+    ARG NAME=macOS-64bit
+  ELSE IF [ "$TARGETPLATFORM" = "darwin/arm64" ]
+    ARG NAME=macOS-arm64
+  ELSE
+    ARG NAME=custom
+  END
+
+  ENV FLATPAK_NAME owncast-$version-$NAME.flatpak
+  COPY (+build/owncast --platform $TARGETPLATFORM)  /build/flatpak/owncast
+  COPY . /build
+  RUN flatpak-builder --privileged --force-clean --repo=owncast-repo build-dir online.owncast.Owncast.json # build flatpak package from JSON file
+  RUN flatpak build-bundle owncast-repo /build/flatpak/owncast.flatpak online.owncast.Owncast # create a single-file bundle of flatpak package
+
+  SAVE ARTIFACT --keep-ts /build/flatpak/owncast.flatpak owncast.flatpak AS LOCAL flatpak/$FLATPAK_NAME
```

Comment:
Since we're saving the rest of the distributable files in `dist` it probably makes sense to put this file in there too.